core/src/main/scala/org/apache/spark/deploy/master/FileSystemPersistenceEngine.scala

```diff
@@ -67,10 +67,10 @@ private[master] class FileSystemPersistenceEngine(
       out = serializer.newInstance().serializeStream(fileOut)
       out.writeObject(value)
     } {
-      fileOut.close()
       if (out != null) {
         out.close()
       }
+      fileOut.close()
```

Comment:
Without this, KryoSerializer complains.
- https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

Reply:
cc @sadikovi